ens-improvement-proposals/ensip-11-evmchain-address-resolution.md

```diff
@@ -0,0 +1,90 @@
+---
+description: >-
+  Introduces coinType for EVM compatible chains (amending ENSIP9).
+---
+
+# ENSIP-9: EVM compatible Chain Address Resolution
+
+| **Author**    | Makoto Inoue \<makoto@ens.domains> |
+| ------------- | -------------------------------- |
+| **Status**    | Draft                            |
+| **Submitted** | 2022-01-13                       |
+
+### Motivation
+
+The existing [ENSIP9(multichain address resolution)](./ensip-9-multichain-address-resolution.md) relies on the existence of coin types on [SLIP44](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) which was designed to define address encoding type for deterministic wallets. As the majority of EVM compatible chains inherit the same encoding type as Ethereum, it is redundant to keep requesting the addition of EVM compatible chains into SLIP 44.
+
+### Specification
+
+This specification amends ENSIP9 by allocating all EVM cointype in the range of 0x800000000, which is currently reserved at SLIP44 as msb (most significant bit) and hence no coin types exist in that range.
+To compute the new coin type for EVM chains, you have to do 0x800000000 | chainId
+
+
+```typescript
+export const convertEVMChainIdToCoinType = (chainId: number) =>{
```

Comment:
Can we provide the inverse function as well?

Reply:
Tried `(0x80000000 | 0 - chainId)  >>> 0` but seems wrong. Do you know what the correct implementation to inverse?